✨ feat(front,design) PLAS-063: Alerts Revamping

bvelitchkine opened the pull request:
# Context

New alerts in accordance with [the design system](https://www.figma.com/file/JpOzqTAvmJ1Ju6x6MZLEU8/LinkedIn-to-Notion?type=design&node-id=48-193&mode=design&t=Fsa7jQWtr01qzjsd-0).

# Solution

When put in alerts, some icons were shrunk. It's the reason why I had to add the `plasmo-shrink-0` class on all icons.

# Priority

High

bvelitchkine committed:
🚚 Alerts are now molecules bc of upcoming icons

bvelitchkine committed:
🎨 New alerts structure

bvelitchkine committed:
🐛 Disable svg shrinkage

bvelitchkine committed:
💄 Re-designed alerts

GauvainThery commented on libs/design-system/src/atoms/icons/svg/ArrowPath.tsx:
```diff
@@ -1,11 +1,12 @@
     width={size}
     height={size}
-    className={className}
+    className={cn('plasmo-shrink-0', className)}
```
Not needed. I'll explain in another comment why 😉
Same for all the other ones

bvelitchkine replied:
Tell me if I forgot something but:
1. I've committed your suggestion below (as you can see)
2. Pulled the code in my local branch
3. Fooled around with the `plasmo-shrink-0` class

Here's what happened:
![image](https://github.com/Bastien-and-Gauvain/monorepo/assets/45038783/fa85f2e0-de56-49aa-8f15-7314aa0be471)
![image](https://github.com/Bastien-and-Gauvain/monorepo/assets/45038783/415d0145-7ce5-4117-a9a0-c918a29e8b3e)

Therefore (except if I'm missing smthg), we need `plasmo-shrink-0`

bvelitchkine replied:
You know what? Let's forget about it. I suggest merging the PR as is, with my plasmo-shrinks everywhere, because, in the end, all will be replaced by the icons from `heroicons`. And as you can see, no unintended behavior with them:
![image](https://github.com/Bastien-and-Gauvain/monorepo/assets/45038783/b5f45166-7dd5-4953-a58f-8907abaa4449)

**First one who's blocked by this pending PR can squash-merge it, as far as I'm concerned!**

GauvainThery replied:
I removed it by putting the Icon into a div 👌 
But aligned with you!

GauvainThery commented on libs/design-system/src/molecules/alerts/alerts.tsx:
```diff
@@ -0,0 +1,74 @@
+const colorMapping = {
+  success: 'plasmo-bg-success plasmo-text-white-transparent90',
+  info: 'plasmo-bg-info plasmo-text-white-transparent90',
+  warning: 'plasmo-bg-pink plasmo-text-white-transparent90',
+  error: 'plasmo-bg-error plasmo-text-white-transparent90',
+};
+
+const iconsMapping = {
+  success: 'CheckCircle' as IconType,
+  info: 'InformationCircle' as IconType,
+  warning: 'ExclamationCircle' as IconType,
+  error: 'ExclamationCircle' as IconType,
+};
+
+const Alert = ({ type = 'info', className, link, message }: AlertsProps) => {
+  return link ? (
+    <a
+      href={link}
+      target="_blank"
+      rel="noreferrer"
+      className={cn(
+        'plasmo-antialiased plasmo-w-80 plasmo-h-20 plasmo-p-3 plasmo-rounded-md plasmo-flex plasmo-flex-row plasmo-items-center',
+        colorMapping[type],
+        className
+      )}>
+      <Icon type={iconsMapping[type]} className="plasmo-m-3" />
+      <p className="plasmo-m-3 plasmo-pt-[1px]">{message}</p>
+    </a>
+  ) : (
+    <span
+      className={cn(
+        'plasmo-antialiased plasmo-w-80 plasmo-h-20 plasmo-p-3 plasmo-rounded-md plasmo-flex plasmo-flex-row plasmo-items-center',
+        colorMapping[type],
+        className
+      )}>
+      <Icon type={iconsMapping[type]} className="plasmo-m-3" />
+      <p className="plasmo-m-3 plasmo-pt-[1px]">{message}</p>
+    </span>
+  );
+};
```
```suggestion
const colorMapping = {
  success: 'plasmo-bg-success plasmo-text-white-transparent90',
  info: 'plasmo-bg-info plasmo-text-white-transparent90',
  warning: 'plasmo-bg-pink plasmo-text-white-transparent90',
  error: 'plasmo-bg-error plasmo-text-white-transparent90',
};

const iconsMapping: Record<string, IconType> = {
  success: 'CheckCircle',
  info: 'InformationCircle',
  warning: 'ExclamationCircle',
  error: 'ExclamationCircle',
};

const Alert = ({ type = 'info', className, link, message }: AlertsProps) => {
  const alert = (
    <div
      className={cn(
        'plasmo-antialiased plasmo-space-x-3 plasmo-font-semibold plasmo-w-full plasmo-p-6 plasmo-leading-4 plasmo-rounded-md plasmo-flex plasmo-flex-row plasmo-items-center',
        colorMapping[type],
        className
      )}>
      <Icon type={iconsMapping[type]} />
      <p className="plasmo-flex-wrap">{message}</p>
    </div>
  );
  return link ? (
    <a href={link} target="_blank" rel="noreferrer">
      {alert}
    </a>
  ) : (
    alert
  );
};
```

big suggestion to facilitate the process but here's some explanations:
1. I don't understand why you cast IconType. Don't hesitate to use Record (ie. Record<string, IconType>)
2. Try not to not duplicate code 👉 it's easier to wrap the main element into an anchor rather than typing it twice
3. Try not to pass fixed width or height to components, we'll handle the size when using the component. And that's why your icon was in struggle. Just don't forget the wrap in this case
4. Don't forget the leading of the text 😉

GauvainThery replied:
5. Use space to separate elements rather than margins

bvelitchkine replied:
1. Didn't know `Record`! Nice one, thank you!
2. You're right, my bad
3. Gotcha. A lot of issues could have been avoided had I used the `space` class instead of margins. Once again: my bad.
4. Duly noted.
5. Yeah of course, my bad

Thanks!

GauvainThery reviewed (changes requested):
See comment 😉

bvelitchkine committed:
Conciseness + minor UI enhancements

Less code duplication, better typing and better tailwind classes for components' layout.

Co-authored-by: GauvainThery <79575726+GauvainThery@users.noreply.github.com>

GauvainThery committed:
Merge branch 'master' into PLAS-63-alerts-revamping

GauvainThery committed:
🎨 remove shrink from icon

GauvainThery reviewed (approved):
LGTM !

GauvainThery committed:
🚨 Remove br
